sdk-core/src/main/java/com/sportradar/unifiedodds/sdk/di/ReadersModule.java

```diff
@@ -66,15 +68,21 @@ private DataProvider<BookmakerDetails> providesProductionDataProvider(LogHttpDat
     }
 
     @Provides @Named("IntegrationDataProvider")
-    private DataProvider<BookmakerDetails> providesIntegrationDataProvider(LogHttpDataFetcher httpDataFetcher,
+    private DataProvider<BookmakerDetails> providesIntegrationDataProvider(SDKInternalConfiguration cfg,
+                                                                           LogHttpDataFetcher httpDataFetcher,
                                                                            @Named("SportsApiJaxbDeserializer") Deserializer deserializer) {
+        EnvironmentSetting setting = EnvironmentManager.getSetting(Environment.Replay);
         return new DataProvider<>(
                 "/users/whoami.xml",
-                EnvironmentManager.getApiHost(Environment.Replay),
+                hostAndPort(setting.getApiHost(), cfg.getAPIPort()),
                 true,
                 Locale.ENGLISH,
                 httpDataFetcher,
                 deserializer
         );
     }
+
+    private String hostAndPort(String host, int port) {
```

Comment:
Apparently the Sports API fails if there is a ":80" in the url for some reason, so if it's 80 then we remove it from the url